### ISSUE TITLE
Odd Error When Collecting Results from ProcessPool

### ISSUE THREAD

**Issue by rhaskett** [noxdafox/pebble#34]
I was testing timeouts on a Win10 machine (pebble version 4.3.9, Python 3.6.6 from conda) trying to collect and print all of the answers that didn't timeout.

    from concurrent.futures import TimeoutError
    from pebble import ProcessPool

    def fibonacci(n):
        if n == 0: return 0
        elif n == 1: return 1
        else: return fibonacci(n - 1) + fibonacci(n - 2)

    def main():
        with ProcessPool() as pool:
            future = pool.map(fibonacci, range(40), timeout=10)
            iterator = future.result()

            all = []
            while True:
                try:
                    all.append(next(iterator))
                except StopIteration:
                    break
                except TimeoutError as e:
                    print(f'function took longer than {e.args[1]} seconds')

            print(all)

It seems to be working but it gives me a very dire error:

```
[0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229, 832040, 1346269, 2178309, 3524578]
RuntimeError: I/O operations still in flight while destroying Overlapped object, the process may crash
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\anaconda3\lib\multiprocessing\spawn.py", line 99, in spawn_main
    new_handle = reduction.steal_handle(parent_pid, pipe_handle)
  File "C:\anaconda3\lib\multiprocessing\reduction.py", line 87, in steal_handle
    _winapi.DUPLICATE_SAME_ACCESS | _winapi.DUPLICATE_CLOSE_SOURCE)
PermissionError: [WinError 5] Access is denied
```

Interestingly, part of the error happens even when `range(40)` is replaced by `range(10)` but only the RuntimeError remains and weirdly there is no traceback given.

> RuntimeError: I/O operations still in flight while destroying Overlapped object, the process may crash

**Comment by noxdafox** [comment]
Hello,

thanks for reporting this.

The issue happens when the Pool is shutdown and does not affect its functionalities. You see the stack trace because the library logs internal problems as well but the crash itself happens in a separate thread and should not affect your application.

I will try to spin a Windows 10 environment and test it as soon as I have some time.

**Comment by rhaskett** [comment]
Thanks.  I've been playing with the concurrent decorator as well on the same win10 system and haven't seen any issues there.

**Comment by noxdafox** [comment]
A more in depth explanation of what might be happening is here: https://bugs.python.org/issue19565

Haven't tried to reproduce it yet.

**Comment by noxdafox** [comment]
I managed to get a Windows machine to try out the issue. It seems the error is correctly handled by `pebble` but yet the crash is shown on the `stderr`.

My suspect is [this line](https://github.com/python/cpython/blob/master/Lib/multiprocessing/process.py#L317). The `print_exc` prints the traceback on `stderr` which would explain why the error seems coming from the Python internals rather than `pebble` modules. `pebble` correctly handles the excepition which is expected to happen when the Pool is shutdown.

I am not sure why the Python community decided to force print the traceback of an exception even when it is expected and handled. I guess the only workaround would be to rework the pool shutdown logic to make it more clean with Windows. This might take a while as it's a quite delicate procedure.

**Comment by noxdafox** [comment]
Version `4.3.10` should address this issue. Could you please verify and confirm? Thanks.

**Comment by noxdafox** [comment]
This should be fixed in `4.3.10` as for #35. Closing the issue. Please feel free to re-open if you are still affected by it.

**Comment by noxdafox** [comment]
FYI, I had to revert the fix as it was causing deadlock issues. I will try to find another way to silence the noise.

**Comment by andrewrsky** [comment]
+1 on this. Having the same issue.

**Comment by noxdafox** [comment]
Fix added in [release 4.6.0](https://github.com/noxdafox/pebble/releases/tag/4.6.0).

Closing this issue, please re-open it if the problem persists.
